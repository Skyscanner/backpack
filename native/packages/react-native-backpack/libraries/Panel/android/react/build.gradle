apply plugin: 'com.android.library'
apply plugin: 'kotlin-android'

android {
  compileSdkVersion rootProject.ext.compileSdkVersion
  buildToolsVersion rootProject.ext.buildToolsVersion


  defaultConfig {
    minSdkVersion rootProject.ext.minSdkVersion
    targetSdkVersion rootProject.ext.targetSdkVersion
    versionCode 1
    versionName "1.0"
  }
}

dependencies{
  implementation "com.facebook.react:react-native:+" // From node_modules
  implementation project(':panel-native')
  implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
}
repositories {
  mavenCentral()
}
