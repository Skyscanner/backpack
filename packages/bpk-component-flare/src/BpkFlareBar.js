/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import PropTypes from 'prop-types';

import { cssModules } from '../../bpk-react-utils';

import STYLES from './bpk-flare-bar.module.scss';
import Pointer from './__generated__/js/pointer';
import CornerRadius from './__generated__/js/corner-radius';

const getClassName = cssModules(STYLES);

const BpkFlareBar = (props) => {
  const { className, rounded, svgClassName, ...rest } = props;

  const classNames = [getClassName('bpk-flare-bar__container')];
  if (className) {
    classNames.push(className);
  }

  const curveClassNames = [getClassName('bpk-flare-bar__curve')];
  const leftCornerRadiusClassNames = [
    getClassName('bpk-flare-bar__rounded-corner'),
  ];
  const rightCornerRadiusClassNames = [
    getClassName('bpk-flare-bar__rounded-corner'),
    getClassName('bpk-flare-bar__rounded-corner--trailing'),
  ];
  if (svgClassName) {
    curveClassNames.push(svgClassName);
    leftCornerRadiusClassNames.push(svgClassName);
    rightCornerRadiusClassNames.push(svgClassName);
  }

  return (
    <div className={classNames.join(' ')} {...rest}>
      <Pointer className={curveClassNames.join(' ')} />
      {rounded && (
        <CornerRadius className={leftCornerRadiusClassNames.join(' ')} />
      )}
      {rounded && (
        <CornerRadius className={rightCornerRadiusClassNames.join(' ')} />
      )}
    </div>
  );
};

BpkFlareBar.propTypes = {
  className: PropTypes.string,
  svgClassName: PropTypes.string,
  rounded: PropTypes.bool,
};

BpkFlareBar.defaultProps = {
  className: null,
  svgClassName: null,
  rounded: false,
};

export default BpkFlareBar;
