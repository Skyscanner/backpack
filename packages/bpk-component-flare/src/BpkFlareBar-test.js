/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2016 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { render } from '@testing-library/react';

import BpkFlareBar from './BpkFlareBar';

describe('BpkFlareBar', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<BpkFlareBar />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with a "className" attribute', () => {
    const { asFragment } = render(<BpkFlareBar className="my-custom-class" />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with a "svgClassName" attribute', () => {
    const { asFragment } = render(
      <BpkFlareBar svgClassName="my-custom-class" />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly with "rounded" set to `false`', () => {
    const { asFragment } = render(<BpkFlareBar rounded={false} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
