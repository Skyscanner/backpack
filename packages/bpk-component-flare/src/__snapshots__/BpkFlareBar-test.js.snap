// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BpkFlareBar should render correctly 1`] = `
<DocumentFragment>
  <div
    class="bpk-flare-bar__container"
  >
    <svg
      class="bpk-flare-bar__curve"
      viewBox="0 0 14832 55"
    >
      <path
        d="M7501.307 8.517l-68.043 39.341c-10.632 6.185-23.795 6.185-34.528 0l-68.144-39.34c-8.91-5.173-18.988-8.215-29.316-8.518H0v55h14832V0H7530.671a63.604 63.604 0 00-29.364 8.517z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`BpkFlareBar should render correctly with "rounded" set to \`false\` 1`] = `
<DocumentFragment>
  <div
    class="bpk-flare-bar__container"
  >
    <svg
      class="bpk-flare-bar__curve"
      viewBox="0 0 14832 55"
    >
      <path
        d="M7501.307 8.517l-68.043 39.341c-10.632 6.185-23.795 6.185-34.528 0l-68.144-39.34c-8.91-5.173-18.988-8.215-29.316-8.518H0v55h14832V0H7530.671a63.604 63.604 0 00-29.364 8.517z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`BpkFlareBar should render correctly with a "className" attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-flare-bar__container my-custom-class"
  >
    <svg
      class="bpk-flare-bar__curve"
      viewBox="0 0 14832 55"
    >
      <path
        d="M7501.307 8.517l-68.043 39.341c-10.632 6.185-23.795 6.185-34.528 0l-68.144-39.34c-8.91-5.173-18.988-8.215-29.316-8.518H0v55h14832V0H7530.671a63.604 63.604 0 00-29.364 8.517z"
      />
    </svg>
  </div>
</DocumentFragment>
`;

exports[`BpkFlareBar should render correctly with a "svgClassName" attribute 1`] = `
<DocumentFragment>
  <div
    class="bpk-flare-bar__container"
  >
    <svg
      class="bpk-flare-bar__curve my-custom-class"
      viewBox="0 0 14832 55"
    >
      <path
        d="M7501.307 8.517l-68.043 39.341c-10.632 6.185-23.795 6.185-34.528 0l-68.144-39.34c-8.91-5.173-18.988-8.215-29.316-8.518H0v55h14832V0H7530.671a63.604 63.604 0 00-29.364 8.517z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
